pfetch-rs: init at 2.7.0

Gobidev opened the pull request:
This PR adds a new package for pfetch-rs, a rewrite of the pfetch system information tool in Rust.

It also adds myself as a maintainer of this package.

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Lord-Valen commented on pkgs/tools/misc/pfetch-rs/default.nix:
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pfetch";
```
```suggestion
  pname = "pfetch-rs";
```

Lord-Valen commented on pkgs/tools/misc/pfetch-rs/default.nix:
```diff
@@ -0,0 +1,23 @@
+
+  src = fetchFromGitHub {
+    owner = "Gobidev";
+    repo = "pfetch-rs";
```
```suggestion
    repo = pname;
```

Lord-Valen reviewed (changes requested):
Please split the change to maintainers/maintainer-list.nix into its own separate commit called `maintainers: add gobidev`.

Two commits total:

`maintainers: add gobidev`
`pfetch-rs: init at 2.7.0`

Gobidev commented:
I added the requested changes

Lord-Valen commented:
Result of `nixpkgs-review pr 237819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pfetch-rs</li>
  </ul>
</details>

Lord-Valen commented:
Works as expected on x86_64-linux.

drupol commented on pkgs/tools/misc/pfetch-rs/default.nix:
```diff
@@ -0,0 +1,29 @@
+    owner = "Gobidev";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0b5300g7q9f07pcr5vy42v6wgb8sk6glai3kb5pj7f9ilq87mmr8";
```
Use `hash` in place of `sha256` please.

Gobidev committed:
maintainers: add gobidev

This commit adds myself as a maintainer

Gobidev committed:
pfetch-rs: init at 2.7.0

This commit adds a new package for pfetch-rs, a rewrite of the pfetch system information tool in Rust

drupol commented:
@GrahamcOfBorg build pfetch-rs
